core/src/main/java/io/questdb/cairo/pool/WriterPool.java

```diff
@@ -226,6 +225,15 @@ public String lock(TableToken tableToken, String lockReason) {
         return reinterpretOwnershipReason(e.ownershipReason);
     }
 
+    public void setCommitListener(TableToken tt, CommitListener listener) {
+        assert commitListeners.get(tt.getTableId()) == null;
```

Comment:
`TableIDs` are not unique enough, there can be 2 tables with the same ID if the user copies a table directory inside QuestDB root directory.

Reply:
> `TableIDs` are not unique enough, there can be 2 tables with the same ID if the user copies a table directory inside QuestDB root directory.

i would call that a hack and we probably should not support that kind of table copies.
we have to assume that table id is unique, otherwise the meaning of table id is lost.
we generate it to serve as a unique identifier for tables.

Reply:
What else can I use then ? 
Would using full tableToken as key be better ?

Reply:
Used TableToken as collection key . hashCode uses tableId while equals checks dirName so it should be fine now .